src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +11,26 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        try {
+            lock.readLock().lock();
+            list.add(element);
+        } finally {
+            lock.readLock().unlock();
```

Comment:
The `add` method should use a write lock instead of a read lock. Use `lock.writeLock().lock()` and `lock.writeLock().unlock()` to ensure thread safety when modifying the list.